### PR TITLE
Add skipped test tracking to JUnit output.

### PR DESCRIPTION
**Pull request by grynspan**
This PR adds skipped test reporting to our JUnit XML output. For example, given this test:

```swift
@Test(.disabled("Because I said so"))
func f() {}
```

The XML output would be, approximately:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="TestResults" errors="0" tests="1" failures="0" skipped="1" time="12345.0">
    <testcase classname="MyTests" name="f()" time="12344.0">
      <skipped>Because I said so</skipped>
    </testcase>
  </testsuite>
</testsuites>
```

See also https://github.com/swiftlang/swift-package-manager/issues/7383 which asks for this for XCTest.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by briancroom** [comment]
Ah nice, thanks!

**Review by briancroom — APPROVED** [review]

File: Sources/Testing/Events/Recorder/Event.JUnitXMLRecorder.swift [hunk 1]
```diff
@@ -183,14 +209,19 @@ extension Event.JUnitXMLRecorder {
   ///
   /// - Returns: `character`, or a string containing its escaped form.
   private static func _escapeForXML(_ character: Character) -> String {
-    if character == #"""# {
+    switch character {
```
Comment (jakepetroules) [inline Sources/Testing/Events/Recorder/Event.JUnitXMLRecorder.swift:212]: `<` and `>` aren't _required_ to be escaped in XML attributes, but `&` definitely is. Could you add that?

Comment (grynspan) [reply Sources/Testing/Events/Recorder/Event.JUnitXMLRecorder.swift:212]: Gonna leave `>` out of pedantry, but yeah I'll add `&`.

**Pull request merged**